internal/server/network/driver_ovn.go

```diff
@@ -1956,14 +1959,54 @@ func (n *ovn) getDHCPv4Reservations() ([]iprange.Range, error) {
 	}
 
 	var dhcpReserveIPv4s []iprange.Range
-	if routerIntPortIPv4 != nil {
-		dhcpReserveIPv4s = []iprange.Range{{Start: routerIntPortIPv4}, {Start: dhcpalloc.GetIP(ipv4Net, -2)}}
+
+	if routerIntPortIPv4 != nil && n.config["ipv4.dhcp.ranges"] != "" {
```

Comment:
That looks wrong.

Prior to this change, any network with IPv4 configured would have its first and last IP addresses reserved. After your change, this reservation now only happens if `ipv4.dhcp.ranges` is non-empty.